src/core/util.js

```diff
@@ -283,4 +285,221 @@ var parseTimeSeriesForC3 = function(graph_data) {
+    var cell;
+    var cell_ref;
+    var ws = {};
+
+    for(R = 0; R < summary_cells.num_rows; ++R) {
+        for(C = 0; C < summary_cells.num_cols; ++C) {
+            cell_ref = XLSX.utils.encode_cell({c:C,r:R});
+            ws[cell_ref] = summary_cells.cells[(R * summary_cells.num_cols) + C];
+        }
+    }
+    for(var data_row = 0; data_row < data_cells.num_rows; ++data_row) {
+        for(var data_col = 0; data_col < data_cells.num_cols; ++data_col) {
+            cell_ref = XLSX.utils.encode_cell({c:data_col,r:R+data_row});
+            ws[cell_ref] = data_cells.cells[(data_row * data_cells.num_cols) + data_col];
+        }
+    }
+    // set combined worksheet range bounds
+    var range = {
+      s: {c:0, r:0}, 
+      e: {
+          c: (summary_cells.num_cols > data_cells.num_cols ? summary_cells.num_cols : data_cells.num_cols), 
+          r: summary_cells.num_rows + data_cells.num_rows + 1}};
+    ws['!ref'] = XLSX.utils.encode_range(range);
+    return ws;
+}
+
+/*
+    Takes current data displayed in the DataTable and contextual data from user input, 
+    creates an XLSX or CSV file, and serves it to the user for download.
+*/
+var exportTableDataToWorksheet = function(data, format) {
+    // create workbook object containing one or more worksheets
+    var wb = {}
+    wb.Sheets = {};
+    wb.SheetNames = [];
+
+    // convert timestep ID (0-16) to string format
+    var times_of_year = ["January", "February", "March", "April", "May", "June", "July", "August", "September", 
+                      "October", "November", "December", "Winter - DJF", "Spring - MAM", "Summer - JJA", 
+                      "Fall - SON", "Annual"];
+    var time_of_year = times_of_year[this.state.dataTableTimeOfYear];
+
+    // prepare summary cells
+    var summary_cells = createWorksheetSummaryCells(this.props, time_of_year);
+
+    // if time_of_year is a season, overwrite it with a short version label for use in output filename (e.g. "Summer - JJA" -> "Summer")
+    var short_season_label = time_of_year.substr(0, time_of_year.indexOf(' ')); // will return string before whitespace
+    if(short_season_label) time_of_year = short_season_label;
+
+    // prepare data cells
+    var data_cells = fillWorksheetDataCells(data);
+
+    // assemble summary_cells and data_cells into one XLSX-encoded worksheet
+    var ws = assembleWorksheet(summary_cells, data_cells);
+
+    // add worksheet to workbook. Note: ws_name will be truncated to 31 chars in XLSX export to meet Excel limitation
+    var ws_name = "Stats_Table_" + this.props.variable_id + "_" + time_of_year;
+    wb.SheetNames.push(ws_name);
+    wb.Sheets[ws_name] = ws;
+
+    // Note: we will probably want to split the rest of this out into a separate function 
+    // to combine multiple worksheets if/when we want to export additional data (e.g. DataGraph points)
+    function to_csv(workbook) {
+        var result = [];
+        workbook.SheetNames.forEach(function(sheetName) {
```

Comment:
Use `_.map(workbook.Sheets, XLSX.utils.sheet_to_csv)` to eliminate the need for `result` and the `result.push()` line.